src/SearchResults.js

```diff
@@ -0,0 +1,80 @@
+import React from 'react';
+import moment from 'moment'
+
+class SearchResults extends React.Component {
+    constructor(props) {
+        super(props);
+        this.state = { isRowSelected: false, selectedRow: null };
+    }
+
+
+
+    highlightRow = (event) => {
+
+        //Check if there is a row selected or not 
+        if (!this.state.isRowSelected && !this.state.selectedRow) {
+
+            //updating the state if a new  row selected and saving the value of the it for later to make sure there is only one row selected 
+            this.setState({
+                isRowSelected: true,
+                selectedRow: event.target.parentNode
+            })
+
+            // adding  class to  highlight the selected row 
+            event.target.parentNode.classList.add("table-danger");
+
+        } else if (this.state.selectedRow === event.target.parentNode) {//here we check if the clicked row is previously selected  if it is it will be unhighlighted
+
+            this.setState({
+                isRowSelected: false,
+                selectedRow: null
+            })
+            // removing the added class in case we clicked a previously highlighted row 
+            event.target.parentNode.classList.remove("table-danger");
```

Comment:
Don't do direct manipulation of the DOM outwith the render method (where you BUILD the DOM). Instead, change the STATE (or props of a child) and let the render() method(s) build the DOM for you.

Reply:
Thank you @nbogie for your feedback, I have updated my code